### DISCUSSION TITLE
Tag editor

### DISCUSSION THREAD

**Discussion by mag01** [losnoco/Cog#254] [Ideas]
Would it be too difficult to implement a tag editor (like for example foobar2000 has)?

**Comment by kode54** [comment]
Tag editing would require implementing both a tag editor dialog, or tag editing into the existing Info dialog. And it would also require implementing tag writing support into the player. Most formats currently supported for tag reading would be implemented by the TagLib writer, which also only handles local files, or over mounted network shares via Finder mounting them. All other formats I support, such as the GME formats, would require manually sculpting tag writers for every format. PSF and all PSF variants would be a lot easier, but still require a tag writer implementation.

I've been told that I shouldn't continue to bloat the player with extra features, but I don't know whether this one would add much noticeable size. It would also give me an excuse to add my R128Gain scanner to the player as a multi-threaded track and album scanner and tagger, similar to what foobar2000 already does now.

I know when I added ReplayGain support to TagLib, I did implement writing for all of the fields for all of the formats that support tag rewriting. So that's already there. Note that it will likely only support the fixed and limited tag fields that are already exposed in the info dialog, and not full key/value editing.

**Reply by brightredspacejoystick** [reply]
well, i'd love to see a tag editor, especially for spc files. i found a OLD spc tag editor (the tags are called id666 😳):

https://www.ninjasftw.com/squirrel/id666/ - which won't run on on a silicon, and i don't readily have an intel to test it on.

but i SO badly wanted to add album art to my playlist today.

(but i'd really only like to see this for video game formats; standard waveform formats (mp3, wav, aiff, etc.) i use nightbirdsevolve's meta (which, your player reminds me of).)
